src/ipfs/encoded.ts

```diff
@@ -0,0 +1,44 @@
+import cbor from 'borc'
+import { CID, FileContent } from './types'
+import basic from './basic'
+import keystore from '../keystore'
+
+export const add = async (content: any, key?: string): Promise<CID> => {
+  const encoded = cbor.encode(content)
+  const toAdd = key !== undefined ? await keystore.encrypt(encoded, key) : encoded 
+  return basic.add(toAdd)
+}
+
+export const catAndDecode = async (cid: CID, key?: string): Promise<any> => {
+  const buf = await basic.catBuf(cid)
+  const toDecode = key !== undefined ? await keystore.decrypt(buf, key) : buf
+  return cbor.decode(toDecode)
+}
+
+export const getBool = async (cid: CID, key?: string): Promise<boolean | undefined> => {
```

Comment:
Can we use a custom type here instead of `undefined`? Would be great to have something along the lines of a `Result`. Something like:
```
type DecodingError = ContentTypeDoesNotMatch
Ipfs.Encoded.getBool(ContentType): Promise<Result<DecodingError, Bool>>
```

Reply:
ended up moving around/changing some of this code. but incorporated your idea around custom error type. I think error handling could be improved in general, gonna think on this...